orttraining/orttraining/training_ops/cuda/optimizer/adam/adam.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <memory>
+#include <utility>
+
+#include "orttraining/training_ops/cuda/optimizer/adam/adam.h"
+#include "orttraining/training_ops/cuda/optimizer/adam/adam_impl.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+ONNX_OPERATOR_KERNEL_EX(
+    Adam,
+    kMSDomain,
+    1,
+    kCudaExecutionProvider,
```

Comment:
Let's add a backlog item to implement this for CPU EP as well. Isn't a high priority right now but we should have this eventually.

Reply:
Done